packages/shared/src/hooks/useAuthForms.ts

```diff
@@ -20,31 +19,45 @@ interface UseAuthFormsProps {
 
 const useAuthForms = ({ onDiscard }: UseAuthFormsProps = {}): UseAuthForms => {
   const [container, setContainer] = useState<HTMLDivElement>();
-  const [isDiscardOpen, setIsDiscardOpen] = useState(false);
+  const { showPrompt } = usePrompt();
   const formRef = useRef<HTMLFormElement>();
 
+  const openPrompt = async (e) => {
+    const options: PromptOptions = {
+      title: 'Discard changes?',
+      description: 'If you leave your changes will not be saved',
+      okButton: {
+        title: 'Leave',
+        className: 'btn-primary-cabbage',
+      },
+      cancelButton: {
+        title: 'Stay',
+        className: 'btn-secondary',
+      },
+      className: {
+        buttons: 'flex-row-reverse',
+      },
+    };
+    if (await showPrompt(options)) {
+      onDiscard(e);
+    }
+  };
+
   const onDiscardAttempt: CloseAuthModalFunc = (e, forceClose = false) => {
     if (forceClose || !formRef?.current) {
       return onDiscard(e, forceClose);
     }
-    return setIsDiscardOpen(true);
-  };
-
-  const onDiscardCanceled: CloseAuthModalFunc = (e) => {
-    e.stopPropagation();
-    setIsDiscardOpen(false);
+    return openPrompt(e);
```

Comment:
@sshanzel I dont think this prompt will be shown anywhere. The only place that uses `onDiscardAttempt` function is already passing forceClose=true in AuthOptions.tsx: 
```
  const onProfileSuccess = async () => {
    await refetchBoot();
    onSuccessfulRegistration?.();
    onClose(null, true);
  };
```

Can we clear up this or am I missing something?

Reply:
I believe this is used in AuthModal.

Reply:
Yeah it's a edge case for the callback on the authForm

Reply:
Thanks, should be fine now.